src/GitVersion.Core/Configuration/BranchConfigurationCalculator.cs

```diff
@@ -34,7 +34,7 @@ private BranchConfig GetBranchConfigurationInternal(int recursions, IBranch targ
             throw new InfiniteLoopProtectionException($"Inherited branch configuration caused {recursions} recursions. Aborting!");
         }
 
-        var matchingBranches = configuration.GetConfigForBranch(targetBranch.Name.WithoutRemote);
+        var matchingBranches = configuration.FindConfigurationForBranch(targetBranch.Name.WithoutRemote);
```

Comment:
Would it be better if we just passed `targetBranch` directly to the method and allowed it to resolve `.Name.WithoutRemote` itself? [I don't like passing around `string` and other primitives if we can avoid it](https://refactoring.guru/smells/primitive-obsession).

Also, since the method sits on the `Config` object, perhaps we can just shorten it to `.ForBranch()`? I think it is clear that what we're asking for is the configuration for a specific branch that way.


```suggestion
        var matchingBranches = configuration.ForBranch(targetBranch);
```

Reply:
Done